pkgs/development/libraries/wlroots/protocols.nix

```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchFromGitLab, pkg-config, wayland-scanner }:
+
+stdenv.mkDerivation rec {
+  pname = "wlr-protocols";
+  rev = "d998ee6fc64ea7e066014023653d1271b7702c09";
+  version = "unstable-2021-11-01";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "wlroots";
+    repo = "wlr-protocols";
+    rev = "${rev}";
+    sha256 = "1vw8b10d1pwsj6f4sr3imvwsy55d3435sp068sj4hdszkxc6axsr";
+  };
+
+  depsBuildBuild = [ pkg-config ];
+  nativeBuildInputs = [ wayland-scanner ];
+
+  installPhase = ''
```

Comment:
I tried to run it without the overriden installPhase but don't get why it fails.
Currently this doesn't install pkg-config files, but upstream does. Is there maybe a way to fix upstream's Makefile?

Reply:
I tried for a while but also gave up - the important thing is the XML. I can try again but no guarantees? It looks like the protocols are pretty static (albeit still marked unstable) so i dont think this repo will get much attention going forward.

The only reason its necessary is for some wlr specific stuff, particularly around efficiently handling output buffers (needed for wl-mirror)

Reply:
Figured it out: behavior invariants with implicit rules on `*.xml`. Added a rule that should address this.